dds/DCPS/security/SSL/DiffieHellman.cpp

```diff
@@ -18,6 +18,15 @@ namespace OpenDDS {
 namespace Security {
 namespace SSL {
 
+namespace {
+  int compare(const DDS::OctetSeq& s1, const char* s2) {
```

Comment:
This can be simplified:
- we don't need to know relative order, just equality/inequality
- if `s1.length()` is not equal to `strlen(s2) + 1`, the strings must not be equal, bail out
- then defer to memcmp